pims/base_frames.py

```diff
@@ -8,6 +8,9 @@
 import numpy as np
 import collections
 import itertools
+import functools
+import contextlib
+from threading import Lock
```

Comment:
The threading stuff came from Dustin who was doing some stuff with threads.   I would say we leave it in and document it as 'probably thread safe'.